code/game/turfs/baseturfs.dm

```diff
@@ -0,0 +1,188 @@
+			target_baseturfs = copytarget.baseturfs.Copy(clamp(1 + ignore_bottom, 1 + copytarget.baseturfs.len - depth, copytarget.baseturfs.len))
+		else if(!ignore_bottom)
+			target_baseturfs = list(copytarget.baseturfs)
+		if(target_baseturfs)
+			target_baseturfs -= new_baseturfs & GLOB.blacklisted_automated_baseturfs
+			new_baseturfs += target_baseturfs
+
+	var/turf/newT = copytarget.copyTurf(src, copy_air)
+	newT.baseturfs = baseturfs_string_list(new_baseturfs, newT)
+	return newT
+
+/// Tries to find the given type in baseturfs.
+/// If found, returns how deep it is for use in other baseturf procs, or null if it cannot be found.
+/// For example, this number can be passed into ScrapeAway to scrape everything until that point.
+/turf/proc/depth_to_find_baseturf(baseturf_type)
+	var/index = baseturfs.Find(baseturf_type)
+	if (index == 0)
+		return null
+	return baseturfs.len - index + 1
+
+/// Returns the baseturf at the given depth.
+/// For example, baseturf_at_depth(1) will give the baseturf that would show up when scraping once.
+/turf/proc/baseturf_at_depth(index)
+	TEST_ONLY_ASSERT(isnum(index), "baseturf_at_depth must be given a number, received [index]")
+	if (islist(baseturfs))
+		return LAZYACCESS(baseturfs, baseturfs.len - index + 1)
+	else if (index == 1)
+		return baseturfs
+	else
+		return null
+
+/// Replaces all instances of needle_type in baseturfs with replacement_type
+/turf/proc/replace_baseturf(needle_type, replacement_type)
+	if (islist(baseturfs))
+		var/list/new_baseturfs
+
+		while (TRUE)
+			var/found_index = baseturfs.Find(needle_type)
+			if (found_index == 0)
+				break
+
+			new_baseturfs ||= baseturfs.Copy()
+			new_baseturfs[found_index] = replacement_type
+
+		if (!isnull(new_baseturfs))
+			baseturfs = baseturfs_string_list(new_baseturfs, src)
+	else if (baseturfs == needle_type)
+		baseturfs = replacement_type
+
+/// Removes all baseturfs that are found in the given typecache.
+/turf/proc/remove_baseturfs_from_typecache(list/typecache)
+	if (islist(baseturfs))
+		var/list/new_baseturfs
+
+		for (var/baseturf in baseturfs)
+			if (!typecache[baseturf])
+				continue
+
+			new_baseturfs ||= baseturfs.Copy()
+			new_baseturfs -= baseturf
+
+		if (!isnull(new_baseturfs))
+			baseturfs = baseturfs_string_list(new_baseturfs, src)
+	else if (typecache[baseturfs])
+		baseturfs = /turf/baseturf_bottom
+
+/// Returns the total number of baseturfs
+/turf/proc/count_baseturfs()
+	return islist(baseturfs) ? length(baseturfs) : 1
+
+/// Inserts a baseturf at the given level.
+/// "Level" here doesn't mean depth.
+/// For example, `insert_baseturf(2, /turf/open/floor/plating)` will make it so
+/// the 2nd to last turf in the list is plating.
+/// This is different from *depth*, since depth is the level from the top.
+/turf/proc/insert_baseturf(level, turf_type)
+	if (!islist(baseturfs))
+		assemble_baseturfs()
+		TEST_ONLY_ASSERT(islist(baseturfs), "baseturfs is not a list after assemble_baseturfs()")
```

Comment:
```suggestion
	if (!islist(baseturfs))
		baseturfs = assemble_baseturfs()
		if(!islist(baseturfs))
			baseturfs = list(baseturfs)
```
assemble_baseturfs is not guarenteed to return a list.
We "resolve" baseturfs in case this is called before init, and then just wrap it in a list so it's usable in just a sec.